src/main/backend.ts

```diff
@@ -163,17 +168,102 @@ export class MainBackend extends BaseBackend<ElectronOptions> implements CommonB
     const hubScope = getCurrentHub().getScope();
     if (hubScope) {
       hubScope.addScopeListener(updatedScope => {
-        const cloned = Scope.clone(updatedScope);
+        const scope = Scope.clone(updatedScope);
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        (cloned as any)._eventProcessors = [];
+        (scope as any)._eventProcessors = [];
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        (cloned as any)._scopeListeners = [];
+        (scope as any)._scopeListeners = [];
+
+        this._scopeStore.set(scope);
 
-        this._scopeStore.set(cloned);
+        // If we use the Crashpad minidump uploader we have to set extra whenever the scope updates
+        //
+        // We do not currently set addExtraParameter on Linux because Breakpad is limited
+        if (this._options.useCrashpadMinidumpUploader === true && process.platform !== 'linux') {
+          this._updateExtraParams(scope);
+        }
       });
     }
   }
 
+  /** Updates Electron uploader extra params */
+  private _updateExtraParams(scope: Scope): void {
+    if (this._updatingExtraParams) {
+      return;
+    }
+
+    this._updatingExtraParams = true;
+
+    forget(
+      (async () => {
+        try {
+          const event = await this._getNativeUploaderEvent(scope);
+          // Update the extra parameters in the main process
+          const mainParams = this._getNativeUploaderExtraParams(event);
+          for (const key of Object.keys(mainParams)) {
+            crashReporter.addExtraParameter(key, mainParams[key]);
+          }
+        } finally {
+          this._updatingExtraParams = false;
+        }
+      })(),
+    );
+  }
```

Comment:
@jan-auer does this look better to avoid race conditions?

Reply:
I'm afraid this could now result in lost updates.

I took a closer look now, and the culprit is that event defaults can be initialized lazily. Changing that is not practical in this PR, so I suggest a small change: Ensure that always the _last_ update comes through instead of the first one. Something like:

```javascript
const currentEpoch = ++this.updateEpoch;

forget(async () => {
  await blablabla();
  if (currentEpoch !== this.updateEpoch) return;
  addExtraParameter(...);
});
```

Reply:
Yeah, this is a much better idea!